src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,185 @@
+                        try {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
+                            startDateTime = LocalDateTime.parse(startDateInput, formatter);
+                            startDate = startDateTime.format(formatter);
+                            break;  // Stop at the first successful parse
+                        } catch (DateTimeParseException e) {
+                            // Ignore the exception and try the next format
+                        }
+                    }
+
+                    // if the input fits none of the format, throw DateTimeParseException
+                    if (startDate == null) {
+                        throw new DateTimeParseException("Invalid date/time format", startDate, 0);
+                    }
+                    LocalDateTime dueDateTime = null;
+                    String dueDate = null;
+
+                    // Try parsing with different formats for due date and time input.
+                    for (String formatString : formatStrings) {
+                        try {
+                            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatString);
+                            dueDateTime = LocalDateTime.parse(dueDateInput, formatter);
+                            dueDate = dueDateTime.format(formatter);
+                            break;  // Stop at the first successful parse
+                        } catch (DateTimeParseException e) {
+                            // Ignore the exception and try the next format
+                        }
+                    }
+
+                    // if the input fits none of the format, throw DateTimeParseException
+                    if (dueDate == null) {
+                        throw new DateTimeParseException("Invalid date/time format", dueDateInput, 0);
+                    }
+
+                    // check if start date is before due date.
+                    if (startDateTime.isAfter(dueDateTime) || startDateTime.isEqual(dueDateTime)) {
+                        throw new IllegalArgumentException();
+                    }
+
+                    // add item into list
+                    Event task = new Event(taskName, startDate, dueDate);
+                    return new AddCommand(task);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    if (input.split(" ").length == 1) {
+                        return new ErrorCommand("☹ OOPS!!! The description of a event cannot be empty");
+                    } else {
+                        return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
+                    }
+                } catch (DateTimeParseException e) {
+                    return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
```

Comment:
This line looks to be over 120 characters, maybe breaking it up into another line would be better?